### PR TITLE
Save conf to dedicated file

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Follow up on #247 with:

- added test
- possibility to pass a formater to use (default to store formater)

cc @mhamann

### CODE REVIEW

**Comment by AdrieanKhisbe** [comment]
@mhamann having played around with the test I feel like we should replace the test framework.
One of the main reason is that vows framework is no longer being supported for years.

As main candidate for replacement I see `ava` and `jest`.
I'm open to take charge of the migration that could be landed with 1.0.

What do you think?

cc @indexzero 

Ps: I can open an issue and/or we could discuss somewhere like https://gitter.im/indexzero/nconf (if the gitter gets activated)

**Comment by mhamann** [comment]
@AdrieanKhisbe I actual just opened an issue on migrating tests to a new framework. It's #282. Feel free to share your thoughts there!

File: test/stores/file-store-test.js [hunk 1]
```diff
@@ -128,6 +129,54 @@ vows.describe('nconf/stores/file').addBatch({
...
+        var that = this,
+            pathFile = '/tmp/nconf-save-toFile.yaml';
+
+        Object.keys(data).forEach(function (key) {
+          tmpStore.set(key, data[key]);
+        });
+
+        tmpStore.saveToFile(pathFile, yamlFormat, function () {
+          fs.readFile(pathFile, function (err, d) {
+            //fs.unlinkSync(pathFile);
```
Comment (mhamann) [inline test/stores/file-store-test.js]: Can you remove this line and the next? I assume it's probably left there from some debugging :-)

Comment (AdrieanKhisbe) [reply test/stores/file-store-test.js]: in fact what I did for debugging was comment the fs.unlink

Comment (AdrieanKhisbe) [reply test/stores/file-store-test.js]: voilà

**Comment by AdrieanKhisbe** [comment]
hey @mhamann, I saw you bump the version to 0.9, did you published it?

**Comment by mhamann** [comment]
Not yet. Hoping to have it out by EOD or so.

**Comment by AdrieanKhisbe** [comment]
Okey dokey ☺

**Comment by AdrieanKhisbe** [comment]
@mhamann busy week? :)

**Comment by AdrieanKhisbe** [comment]
@mhamann I notice few glitch by starting to integrating latest changes in my nconf library.
I'm gonna submit a little lit of PR that might be interesting to consider for 0.9.

This starts with #288

**Comment by mhamann** [comment]
@AdrieanKhisbe 0.9.0 is released to npm. Will address bugs in 0.9.1.

**Pull request merged**